probe-rs/src/architecture/arm/core/exception_handling/armv7m.rs

```diff
@@ -268,54 +273,102 @@ impl ExceptionReason {
                 {
                     Ok(source)
                 } else {
-                    Ok("MemManage Fault handler. Cause: Unknown.".to_string())
+                    Ok("UsageFault <Cause: Unknown>".to_string())
                 }
             }
-            ExceptionReason::SVCall => Ok("Supervisor call.".to_string()),
-            ExceptionReason::DebugMonitor => Ok("Synchronous Debug monitor fault.".to_string()),
-            ExceptionReason::PendSV => Ok("Pending Supervisor call.".to_string()),
-            ExceptionReason::SysTick => Ok("Systick handler.".to_string()),
-            ExceptionReason::ExternalInterrupt(exti) => Ok(format!("External interrupt #{exti}.")),
+            ExceptionReason::SVCall => Ok("SVC".to_string()),
+            ExceptionReason::DebugMonitor => Ok("DebugMonitor".to_string()),
+            ExceptionReason::PendSV => Ok("PendSV".to_string()),
+            ExceptionReason::SysTick => Ok("SysTick".to_string()),
+            ExceptionReason::ExternalInterrupt(exti) => Ok(format!("External interrupt #{exti}")),
             ExceptionReason::Reserved => {
-                Ok("Reserved by the ISA, and not usable by software.".to_string())
+                Ok("<Reserved by the ISA, and not usable by software>".to_string())
             }
         }
     }
+
+    /// Determines how the exception return address should be offset when unwinding the stack.
+    /// See Armv7-M Architecture Reference Manual, Section B1.5.6.
+    pub(crate) fn is_precise_fault(&self, memory: &mut dyn MemoryInterface) -> Result<bool, Error> {
```

Comment:
From the description I am not sure I understand what this does. Would you mind precising?

Also, could this be a trait method?

Reply:
I wouldn't make it a trait, It is very Cortex-M specific.